Rubberduck.VBEEditor/ComManagement/ProjectsRepository.cs

```diff
@@ -274,6 +274,21 @@ public IVBComponent Component(QualifiedModuleName qualifiedModuleName)
             return EvaluateWithinReadLock(() => _components.TryGetValue(qualifiedModuleName, out var component) ? component : null);
         }
 
+        public void RemoveComponent(QualifiedModuleName qualifiedModuleName)
+        {
+            ExecuteWithinWriteLock(() =>
+            {
+                if (!_components.TryGetValue(qualifiedModuleName, out var component) ||
+                    !_componentsCollections.TryGetValue(qualifiedModuleName.ProjectId, out var components))
+                {
+                    return;
+                }
+
+                _components.Remove(qualifiedModuleName);
+                components.Remove(component);                
```

Comment:
having `components` and `_components` seems like a recipe for desaster...

Reply:
It was, hence the AV :-). Will try and find a better name...

Reply:
renamed and commented